### ISSUE TITLE
Question about the depth loss

### ISSUE THREAD

**Issue by Liuyveg** [zju3dv/manhattan_sdf#25]
Hi, thanks for your beautiful work. And I recently read your paper, it's so amazing and enlightening. When viewing your source code, I'm confused about the `depth_loss_clamp`.
```
 if 'depth_loss_clamp' in loss_weights:
                depth_loss = depth_loss.clamp(max=loss_weights['depth_loss_clamp'])
```
Why the depth loss should be clamped?

**Comment by ghy0324** [comment]
Hi! Thanks for your interest! This is because we noticed that the depth supervision from COLMAP can be very noisy.
As shown in the code, we clamp the depth loss after training for some iterations.
https://github.com/zju3dv/manhattan_sdf/blob/604944fd37cd8eb9045f4eb8ec4f994269981af4/lib/train/trainers/trainer.py#L54-L55
In general, the geometry maybe not very accurate, but won’t be too far from ground truth after training for some iterations. So if some pixels are of too large depth loss, it is mostly likely because the noise of COLMAP depth. We clamp the depth loss to alleviate the misleading of noises.

**Comment by Liuyveg** [comment]
I see, thanks for your kind interpretation.
